### ISSUE TITLE
Unit tests!

### ISSUE THREAD

**Issue by ajunge** [uport-project/3box-js#3]
No tests. We need to write unit test. Jest ?

**Comment by oed** [comment]
Take a look at this branch: https://github.com/uport-project/3box-js/tree/feature/implement-private-store

**Comment by ajunge** [comment]
Looks good. Any reason not to mockup ipfs-api and instead spawn an ipfs process?

**Comment by oed** [comment]
Yeah, it's a **lot** more work. Especially with orbitdb, don't know exactly what methods it is using. 
Is there a reason to **not** spawn an ipfs process?

**Comment by ajunge** [comment]
The idea of unit test is to isolate the code from outside factors. I think we can mockup ipfs orbit-db or any code that is not part of our code

**Comment by oed** [comment]
I'd say that the purpose of a unit test is to test if the specific unit works as intended, regardless of how the entire system works. The purpose of the "store" units is to communicate correctly with ipfs and orbitdb. So if we mock up these modules and do some small minor error (which is very likely) we'll end up building a unit that works great with our mocks but not with the actual modules we are using.

ipfsd-cli is built for this purpose, to test code isolated from any real ipfs node (such as infura). So in a sense we are using a mock ipfs instead of the real api we use in production.
